cmd/server.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_zap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
+	grpc_validator "github.com/grpc-ecosystem/go-grpc-middleware/validator"
+	"github.com/spf13/cobra"
+	"github.com/tinkerbell/pbnj/pkg/logging/zaplog"
+	"github.com/tinkerbell/pbnj/pkg/service/grpcsvc"
```

Comment:
the concern I have here is that both of these packages don't have interfaces. if pbnj utilizes them, it will be tightly coupled throughout its codebase, and later when we move to something like `logr` there will be a lot of unnecessary effort to make the change. if these packages had interfaces with specific implementations to use, that would be more ideal. That's the advantage of `logr`. we can create specific/opinionated implementations for Tinkerbell or Packet and require very little effort to swap them in and out.

with all that said, I've added a `packethost/pkg/log` implementation of `logr` to this PR (`pbnj/pkg/logging/packetlog`). It works and I don't have to change any code in `pbnj/pkg/service`, which is great. good decoupling there. Just some ideas/brainstorming. Let me know what you think.

Dave Cheney explains this idea way better than me [here](https://dave.cheney.net/2016/08/20/solid-go-design), under the sections `Interface Segregation Principle` and `Dependency Inversion Principle`.

Reply:
wonderful! PBnJ continuing the tradition of moving our logging setup forward :p. Now throw the `logr` implementation into packethost/pkg :D.

Reply:
copy, PR: https://github.com/packethost/pkg/pull/33

Reply:
once that PR goes through I'll update here to pull that in.